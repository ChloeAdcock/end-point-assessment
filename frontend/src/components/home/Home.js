import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import MapContainer from "../mapContainer/MapContainer";
import MyEvents from '../myEvents/MyEvents';
import Typography from "@material-ui/core/Typography";
import Alert from "@material-ui/lab/Alert";
import { useHistory } from "react-router";
import { getEvents, closeAlert } from "../../redux/actions/events/events";
import { logout } from "../../redux/actions/accounts/accounts";;

function AllEvents() {
  const dispatch = useDispatch();
  const history = useHistory();
  const events = useSelector((state) => state.events.events);
  const errorState = useSelector((state) => state.events.viewAllError);

  const mapStyles = {
    height: "92vh",
    width: "100%",
  };

  const handleClick = (selected) => {
    history.push({
      pathname: "/eventdetails",
      state: selected,
    });
  };

  const handleClose = () => {
    dispatch(closeAlert());
    dispatch(logout());
  };

  useEffect(() => {
    dispatch(getEvents());
  }, [dispatch]);

  if (errorState) {
    return (
      <Alert severity="error" onClose={handleClose}>
        An error has occurred - please try again later
      </Alert>
    );
  } else if (!events) {
    return <Typography>Loading...</Typography>
  }else if (events.length < 1) {
    return <Typography>No events found</Typography>;
  } else {
    return (
      <div>
        <MyEvents />
          <MapContainer
            mapStyles={mapStyles}
            events={events}
            handleClick={handleClick}
            centre={{
              lat: 52.6309,
              lng: 1.2974,
            }}
          />
      </div>
    );
  }
}

export default AllEvents;
