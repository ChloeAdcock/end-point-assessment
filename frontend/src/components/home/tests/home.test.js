import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Home from "../Home";

jest.mock("../../mapContainer/MapContainer", () => {
    return function MockMap(props) {
      return (
          <div>Test</div>
      );
    };
  });

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe("The home component", () => {
    it("should render the map when there are events in the state and no error", () => {
      const store = mockStore({
        events: {
          events: ['test'],
          viewAllError: false,
        },
      });
      const tree = renderer
        .create(
          <Provider store={store}>
            <Home />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render an error alert when there is an error in the state", () => {
        const store = mockStore({
          events: {
            viewAllError: true,
          },
        });
        const tree = renderer
          .create(
            <Provider store={store}>
              <Home />
            </Provider>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it("should render no events found when there are no results and there is no error", () => {
        const store = mockStore({
          events: {
            events: [],
          },
        });
        const tree = renderer
          .create(
            <Provider store={store}>
              <Home />
            </Provider>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it("should render a loading screen when events have not loaded yet", () => {
        const store = mockStore({
            events: {
                events: null
            },
        });
        const tree = renderer
          .create(
            <Provider store={store}>
              <Home />
            </Provider>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
  
  });