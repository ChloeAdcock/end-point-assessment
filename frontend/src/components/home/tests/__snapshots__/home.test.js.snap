// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The home component should render a loading screen when events have not loaded yet 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  Loading...
</p>
`;

exports[`The home component should render a loading screen when there are no events 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  Loading...
</p>
`;

exports[`The home component should render an error alert when there is an error in the state 1`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
  role="alert"
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  >
    An error has occurred - please try again later
  </div>
  <div
    className="MuiAlert-action"
  >
    <button
      aria-label="Close"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Close"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`The home component should render no events found when there are no results and there is no error 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  No events found
</p>
`;

exports[`The home component should render the map when there are events in the state and no error 1`] = `
<div>
  Test
</div>
`;
